source/rhoas/src/main/java/com/openshift/cloud/controllers/AbstractCloudServicesController.java

```diff
@@ -52,13 +56,37 @@
         finished.setMessage(t.getMessage());
         finished.setStatus(Status.False);
       }
-
-      return UpdateControl.updateStatusSubResource(resource);
+      if (!updateLabels) {
+        return UpdateControl.updateStatusSubResource(resource);
+      } else {
+        return UpdateControl.updateCustomResourceAndStatus(resource);
+      }
     } else {
       return UpdateControl.noUpdate();
     }
   }
 
+  private boolean requiresLabelUpdate(T resource) {
+    var updateRequired = false;
+    if (resource instanceof KafkaConnection) {
+      var labels = resource.getMetadata().getLabels();
+
+      if (labels == null) {
+        resource.getMetadata().setLabels(labels = new HashMap());
+      }
+
+      updateRequired = !(labels.containsKey("app.kubernetes.io/component")
```

Comment:
managed-by can already exist as it was documented before and used in UI. 
So my take is that we should check for `app.kubernetes.io/component` and force two flags for it.